File: shared/common/src/main/java/com/visor/school/common/api/ValidationErrorResponse.java
package com.visor.school.common.api;

import java.time.Instant;
import java.util.List;

/**
 * Validation error response for input validation failures.
 * Provides detailed field-level error information.
 */
public record ValidationErrorResponse(
        Instant timestamp,
        int status,
        String error,
        String message,
        String path,
        String correlationId,
        List<FieldError> fieldErrors,
        List<String> globalErrors
) {
    public ValidationErrorResponse(Instant timestamp, int status, String path) {
        this(timestamp, status, "Validation Error", "Input validation failed", path, null, List.of(), List.of());
    }

    public record FieldError(String field, Object rejectedValue, String message, String code) {
        public FieldError(String field, Object rejectedValue, String message) {
            this(field, rejectedValue, message, null);
        }
    }
}

File: services/common/src/main/kotlin/com/visor/school/common/api/ValidationErrorResponse.kt
package com.visor.school.common.api

import java.time.Instant

/**
 * Validation error response for input validation failures
 * Provides detailed field-level error information
 */
data class ValidationErrorResponse(
    val timestamp: Instant = Instant.now(),
    val status: Int,
    val error: String = "Validation Error",
    val message: String = "Input validation failed",
    val path: String,
    val correlationId: String? = null,
    val fieldErrors: List<FieldError> = emptyList(),
    val globalErrors: List<String> = emptyList()
) {
    data class FieldError(
        val field: String,
        val rejectedValue: Any?,
        val message: String,
        val code: String? = null
    )
}

